File: android/src/main/kotlin/com/example/scgateway_flutter_plugin/models/RatiosDTO.kt
package com.example.scgateway_flutter_plugin.models

import com.google.gson.annotations.SerializedName

data class RatiosDTO(

        @SerializedName("52wHigh")
        val jsonMember52wHigh: Any? = null,

        @SerializedName("52wLow")
        val jsonMember52wLow: Any? = null,

        @SerializedName("divYield")
        val divYield: Any? = null,

        @SerializedName("divYieldDifferential")
        val divYieldDifferential: Any? = null,

        @SerializedName("largeCapPercentage")
        val largeCapPercentage: Any? = null,

        @SerializedName("marketCapCategory")
        val marketCapCategory: String? = null,

        @SerializedName("midCapPercentage")
        val midCapPercentage: Any? = null,

        @SerializedName("pb")
        val pb: Any? = null,

        @SerializedName("pbDiscount")
        val pbDiscount: Any? = null,

        @SerializedName("pe")
        val pe: Any? = null,

        @SerializedName("peDiscount")
        val peDiscount: Any? = null,

        @SerializedName("smallCapPercentage")
        val smallCapPercentage: Any? = null,

        @SerializedName("cagr")
        val cagr: Any? = null,

        @SerializedName("momentumRank")
        val momentumRank: Any? = null,

        @SerializedName("risk")
        val risk: Any? = null,

        @SerializedName("sharpe")
        val sharpe: Any? = null,

        @SerializedName("ema")
        val ema: Any? = null,

        @SerializedName("momentum")
        val momentum: Any? = null,

        @SerializedName("lastCloseEma")
        val lastCloseEma: Any? = null,

        @SerializedName("sharpeRatio")
        val sharpeRatio: Any? = null
)

File: example/lib/app/models/RatiosDTO.dart

class RatiosDTO {

  final dynamic jsonMember52wHigh;

  final dynamic jsonMember52wLow;

  final dynamic divYield;

  final dynamic divYieldDifferential;

  final dynamic largeCapPercentage;

  final dynamic marketCapCategory;

  final dynamic midCapPercentage;

  final dynamic pb;

  final dynamic pbDiscount;

  final dynamic pe;

  final dynamic peDiscount;

  final dynamic smallCapPercentage;

  final dynamic cagr;

  final dynamic momentumRank;

  final dynamic risk;

  final dynamic sharpe;

  final dynamic ema;

  final dynamic momentum;

  final dynamic lastCloseEma;

  final dynamic sharpeRatio;

  RatiosDTO({
    this.jsonMember52wHigh,
    this.jsonMember52wLow,
    this.divYield,
    this.divYieldDifferential,
    this.largeCapPercentage,
    this.marketCapCategory,
    this.midCapPercentage,
    this.pb,
    this.pbDiscount,
    this.pe,
    this.peDiscount,
    this.smallCapPercentage,
    this.cagr,
    this.momentumRank,
    this.risk,
    this.sharpe,
    this.ema,
    this.momentum,
    this.lastCloseEma,
    this.sharpeRatio
});

  factory RatiosDTO.fromJson(Map<String, dynamic> parsedJson) {

    return RatiosDTO(
      jsonMember52wHigh: parsedJson['jsonMember52wHigh'],
      jsonMember52wLow: parsedJson['jsonMember52wLow'],
      divYield: parsedJson['divYield'],
      divYieldDifferential: parsedJson['divYieldDifferential'],
      largeCapPercentage: parsedJson['largeCapPercentage'],
      marketCapCategory: parsedJson['marketCapCategory'],
      midCapPercentage: parsedJson['midCapPercentage'],
      pb: parsedJson['pb'],
      pbDiscount: parsedJson['pbDiscount'],
      pe: parsedJson['pe'],
      peDiscount: parsedJson['peDiscount'],
      smallCapPercentage: parsedJson['smallCapPercentage'],
      cagr: parsedJson['cagr'],
      momentumRank: parsedJson['momentumRank'],
      risk: parsedJson['risk'],
      sharpe: parsedJson['sharpe'],
      ema: parsedJson['ema'],
      momentum: parsedJson['momentum'],
      lastCloseEma: parsedJson['lastCloseEma'],
      sharpeRatio: parsedJson['sharpeRatio']
    );

  }
}